src/Xamarin.Android.Build.Tasks/Utilities/RtxtParser.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks
+{
+	public enum RType {
+		Integer,
+		Array,
+	}
+	public enum ResourceType {
+		System,
+		Custom,
+	}
+	public struct R : IComparable<R> {
+		public RType Type;
+		public int Id;
+		public int [] Ids;
+		public string Identifier;
+		public string ResourceTypeName;
+		public ResourceType ResourceType;
+
+		public string Key => $"{ResourceTypeName}:{Identifier}";
+
+		public override string ToString () {
+			if (Type == RType.Integer) {
+				if (ResourceTypeName == "styleable")
+					return $"int {ResourceTypeName} {Identifier} {Id}";
+				return $"int {ResourceTypeName} {Identifier} 0x{Id.ToString ("x8")}";
+			}
+			return $"int[] {ResourceTypeName} {Identifier} {{ {String.Join (", ", Ids.Select (x => $"0x{x.ToString ("x8")}"))} }}";
+		}
+
+		public string ToSortedString () {
+			return $"{ResourceTypeName}_{Identifier}";
+		}
+
+		public int CompareTo(R other) {
+			return String.Compare (ToSortedString (), other.ToSortedString (), StringComparison.OrdinalIgnoreCase);
+		}
+
+		public void UpdateId (int newId) {
+			Id = newId;
+		}
+
+		public void UpdateIds (int [] newIds)
+		{
+			Ids = newIds;
+		}
+	}
+
+	public class RtxtParser {
+
+		TaskLoggingHelper log;
+		Dictionary<string, string> map;
+
+		public static HashSet<string> knownTypes = new HashSet<string> () {
+			"anim",
+			"animator",
+			"attr",
+			"array",
+			"bool",
+			"color",
+			"dimen",
+			"drawable",
+			"font",
+			"id",
+			"integer",
+			"interpolator",
+			"layout",
+			"menu",
+			"mipmap",
+			"plurals",
+			"raw",
+			"string",
+			"style",
+			"styleable",
+			"transition",
+			"xml",
+		};
+
+		public IEnumerable<R> Parse (string file, TaskLoggingHelper logger, Dictionary<string, string> mapping){
+			log = logger;
+			map = mapping;
+			var result = new List<R> ();
+			if (File.Exists (file))
+				ProcessRtxtFile (file, result);
+			return result;
+		}
+
+		void ProcessRtxtFile (string file, IList<R> result)
+		{
+			var lines = File.ReadLines (file);
+			int lineNumber = 0;
+			foreach (var line in lines) {
+				lineNumber++;
+				var items = line.Split (new char [] { ' ' }, 4);
+				if (items.Length < 4) {
+					log.LogDebugMessage ($"'{file}:{lineNumber}' ignoring contents '{line}', it does not have the correct number of elements.");
+					continue;
+				}
+				int value = items [1] != "styleable" ? Convert.ToInt32 (items [3], 16) : -1;
+				string itemName = ResourceIdentifier.GetResourceName(items [1], items [2], map, log);
+				if (knownTypes.Contains (items [1])) {
+					if (items [1] == "styleable") {
```

Comment:
I think this would be clearer if we *inverted* the logic here:

```csharp
if (items [1] != "styleable") {
    results.Add (new R {
        … 
    });
    continue;
}
switch (items [0]) {
// …
}
```